### PR TITLE
provide guidelines for avoiding CVE-2019-3016

### LINKED ISSUE #1583: Update documentation with guidelines for CVE-2019-3016
In a Linux KVM guest that has PV TLB enabled, a process in the guest kernel may be able to read memory locations from another process in the same guest. This problem is limited to the host running linux kernel 4.10 with a guest running linux kernel 4.16 or later. The problem mainly affects AMD processors but Intel CPUs cannot be ruled out.

This vulnerability is described in [CVE-2019-3016](https://nvd.nist.gov/vuln/detail/CVE-2019-3016).

We have to update the [prod-host-seup.md](https://github.com/firecracker-microvm/firecracker/blob/master/docs/prod-host-setup.md) in order to offer guidelines for this vulnerability.

### PR DESCRIPTION
**Pull request by serban300**
## Reason for This PR

Fixes #1583

## Description of Changes

Provided in the commit message.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria. Where there are two options, keep one.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [ ] All commits in this PR are signed (`git commit -s`).
- [ ] Either this PR is linked to an issue, or, the reason for this PR is
      clearly provided.
- [ ] The description of changes is clear and encompassing.
- [ ] Either no docs need to be updated as part of this PR, or, the required
      doc changes are included in this PR. Docs in scope are all `*.md` files
      located either in the repository root, or in the `docs/` directory.
- [ ] Either no code has been touched, or, code-level documentation for touched
      code is included in this PR.
- [ ] Either no API changes are included in this PR, or, the API changes are
      reflected in `firecracker/swagger.yaml`.
- [ ] Either the changes in this PR have no user impact, or, the changes in
      this PR have user impact and have been added to the `CHANGELOG.md` file.
- [ ] Either no new `unsafe` code has been added, or, the newly added `unsafe`
      code is unavoidable and properly documented.

### CODE REVIEW

File: docs/prod-host-setup.md [hunk 1]
```diff
@@ -167,3 +167,35 @@ Verify that swap is disabled by running:
 ```bash
 grep -q "/dev" /proc/swaps && echo "swap partitions present (Recommendation: no swap)" || echo "no swap partitions (OK)"
 ```
+
+#### Use a kernel that is patched for [CVE-2019-3016](https://nvd.nist.gov/vuln/detail/CVE-2019-3016)
```
Comment (raduiliescu) [inline docs/prod-host-setup.md]: I don't know if we should put this here. This is a transitory state, until the fix for this issue gets merged in stable kernels and in long term support distros.
As a security practice anyone should run on updated kernels.

I would just post a security notice.

File: docs/prod-host-setup.md [hunk 2]
```diff
@@ -167,3 +167,35 @@ Verify that swap is disabled by running:
 ```bash
 grep -q "/dev" /proc/swaps && echo "swap partitions present (Recommendation: no swap)" || echo "no swap partitions (OK)"
 ```
+
```
Comment (raduiliescu) [inline docs/prod-host-setup.md]: I would make a section for Known kernel issues
Recommend to keep host kernel up to date.

Put there a recommendation for this CVE with following format:
Issue Description
 mention we would close this issue once solved in stable kernels
Impact
Vulnerable systems - checklist on how to find if a system is vulnerable
Mitigation
 if vulnerable update kernel once patch is provided
Recommendations if not vulnerable - there say to not update kernels...

Comment (serban300) [reply docs/prod-host-setup.md]: Done.

**Comment by serban300** [comment]
I did a smal lchange in order to mention that the host has to be updated when possible. 

@aghecenco @raduiliescu  please take another look !

**Pull request merged**